test/client-unit/services/StockService.spec.js

```diff
@@ -0,0 +1,126 @@
+/* global inject, expect */
+
+describe('StockService', StockServiceTests);
+
+function StockServiceTests() {
+  // define modules
+  beforeEach(module(
+    'bhima.services',
+    'bhima.constants',
+    'ngStorage',
+    'tmh.dynamicLocale',
+    'angularMoment',
+    'pascalprecht.translate',
+  ));
+
+  let Stock;
+
+  // inject dependencies
+  beforeEach(inject((_StockService_) => {
+    Stock = _StockService_;
+  }));
+
+  it('#statusLabelMap() returns the correct stock status label', () => {
+    const status = {
+      sold_out          : 'STOCK.STATUS.SOLD_OUT',
+      in_stock          : 'STOCK.STATUS.IN_STOCK',
+      security_reached  : 'STOCK.STATUS.SECURITY',
+      minimum_reached   : 'STOCK.STATUS.MINIMUM',
+      over_maximum      : 'STOCK.STATUS.OVER_MAX',
+    };
+    expect(Stock.statusLabelMap('sold_out')).to.be.equal(status.sold_out);
+    expect(Stock.statusLabelMap('in_stock')).to.be.equal(status.in_stock);
+    expect(Stock.statusLabelMap('security_reached')).to.be.equal(status.security_reached);
+    expect(Stock.statusLabelMap('minimum_reached')).to.be.equal(status.minimum_reached);
+    expect(Stock.statusLabelMap('over_maximum')).to.be.equal(status.over_maximum);
+  });
+
+  it('#processLotsFromStore() returns a flat array of lots from StockForm.store.data', () => {
+    const entityUuid = 'xxxe7870-0c8e-47d4-a7a8-a17a9924bxxx';
+
+    const mockStockFormStoreData = [
+      {
+        inventory_uuid : 'd03e7870-0c8e-47d4-a7a8-a17a9924b3f4',
+        unit_cost : 0.5,
+        lots : [
+          {
+            uuid : 'd03e7870-0c8e-47d4-a7a8-a17a9924b123', lot : 'A', quantity : 50, expiration_date : '2019-01-01',
+          },
+          {
+            uuid : 'd03e7870-0c8e-47d4-a7a8-a17a9924b124', lot : 'B', quantity : 30, expiration_date : '2018-12-01',
+          },
+        ],
+      },
+    ];
+
+    const expectedLotsArray = [
+      {
+        uuid : mockStockFormStoreData[0].lots[0].uuid,
```

Comment:
An important part of tests is readability - and this isn't very readable to me.  I propose that we clean this up by creating some extra variables.  Like so:

```js
const lotA = {
  uuid : 'd03e7870-0c8e-47d4-a7a8-a17a9924b123',
  lot : 'A',
  quantity : 50,
  expiration_date : '2019-01-01',
};

const lotB = {
  uuid : 'd03e7870-0c8e-47d4-a7a8-a17a9924b124',
  lot : 'B',
  quantity : 30,
  expiration_date : '2018-12-01',
};
```
Now that you have those, you can do this:
```js
// a singe inventory with two lots
const mockStockFormStoreData = [{
  inventory_uuid : 'd03e7870-0c8e-47d4-a7a8-a17a9924b3f4',
  unit_cost : 0.5,
  lots : [ lotA, lotB ],
}];

const processed = Stock.processLotsFromStore(mockStockFormStoreData, entityUuid);
const [first] = processed;

const result = {
  uuid : 'd03e7870-0c8e-47d4-a7a8-a17a9924b123',
  lot : 'A',
  quantity : 50,
  expiration_date : '2019-01-01',
  inventory_uuid : 'd03e7870-0c8e-47d4-a7a8-a17a9924b3f4',
  unit_cost : 0.5,
  origin_uuid : entityUuid,
};

expect(first).to.deep.equal(result);
```